Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/CoilUtils.kt

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package org.dolphinemu.dolphinemu.utils
+
+import android.net.Uri
+import android.view.View
+import android.widget.ImageView
+import coil.load
+import coil.target.ImageViewTarget
+import org.dolphinemu.dolphinemu.R
+import org.dolphinemu.dolphinemu.adapters.GameAdapter.GameViewHolder
+import org.dolphinemu.dolphinemu.features.settings.model.BooleanSetting
+import org.dolphinemu.dolphinemu.model.GameFile
+import java.io.File
+import java.io.FileNotFoundException
+
+object CoilUtils {
+    fun loadGameCover(
+        gameViewHolder: GameViewHolder?,
+        imageView: ImageView,
+        gameFile: GameFile,
+        customCoverUri: Uri?
+    ) {
+        imageView.scaleType = ImageView.ScaleType.FIT_CENTER
+        val imageTarget = ImageViewTarget(imageView)
+        if (customCoverUri != null) {
+            imageView.load(customCoverUri) {
+                error(R.drawable.no_banner)
+                target(
+                    onSuccess = { success ->
+                        disableInnerTitle(gameViewHolder)
+                        imageTarget.drawable = success
+                    },
+                    onError = { error ->
+                        enableInnerTitle(gameViewHolder)
+                        imageTarget.drawable = error
+                    }
+                )
+            }
+        } else if (BooleanSetting.MAIN_USE_GAME_COVERS.booleanGlobal) {
+            imageView.load(CoverHelper.buildGameTDBUrl(gameFile, CoverHelper.getRegion(gameFile))) {
+                error(R.drawable.no_banner)
+                target(
+                    onSuccess = { success ->
+                        disableInnerTitle(gameViewHolder)
+                        imageTarget.drawable = success
+                    },
+                    onError = { error ->
+                        if (gameFile.region == 2) {
```

Comment:
Please add a named constant in GameFile.java for this 2. The code is hard to follow otherwise.